kafka-0.10.x/src/main/scala/monix/kafka/KafkaConsumerObservable.scala

```diff
@@ -62,44 +65,61 @@ trait KafkaConsumerObservable[K, V, Out] extends Observable[Out] {
   private def feedTask(out: Subscriber[Out]): Task[Unit] = {
     Task.create { (scheduler, cb) =>
       implicit val s = scheduler
-      val feedTask = consumer.flatMap { c =>
-        // Skipping all available messages on all partitions
-        if (config.observableSeekOnStart.isSeekEnd) c.seekToEnd(Nil.asJavaCollection)
-        else if (config.observableSeekOnStart.isSeekBeginning) c.seekToBeginning(Nil.asJavaCollection)
-        // A task to execute on both cancellation and normal termination
-        val onCancel = cancelTask(c)
-        runLoop(c, out).guarantee(onCancel)
-      }
-      feedTask.runAsync(cb)
+      val startConsuming =
+        consumerT.bracket { c =>
+          // Skipping all available messages on all partitions
+          if (config.observableSeekOnStart.isSeekEnd) c.seekToEnd(Nil.asJavaCollection)
+          else if (config.observableSeekOnStart.isSeekBeginning) c.seekToBeginning(Nil.asJavaCollection)
+          Task.race(runLoop(c, out), pollHeartbeat(c).loopForever).void
+        } { consumer =>
+          // Forced asynchronous boundary
+          Task.evalAsync(consumer.synchronized(blocking(consumer.close()))).memoizeOnSuccess
```

Comment:
`memoize` seems to be redundant, is there any situation where the finalizer would be called more than once?

Reply:
Totally